internal/services/network/route_server_resource.go

```diff
@@ -153,7 +158,7 @@ func resourceRouteServerCreateUpdate(d *pluginsdk.ResourceData, meta interface{}
 		ContinuousTargetOccurence: 5,
 		Timeout:                   time.Until(timeout),
 	}
-	_, err := stateConf.WaitForStateContext(ctx)
+	_, err = stateConf.WaitForStateContext(ctx)
```

Comment:
is this still required if we poll on the future?

Reply:
Thanks @tombuildsstuff for the comments, yes, this is still needed as we need to get the resource provisioningStatus to ensure the resource is in a ready state.